### PR TITLE
Infoblox parser updates

### PR DESCRIPTION
**Pull request by rushriva**
Parser updates

   Required items, please complete
   
   Change(s):
   - Infoblox parser query updated
   - Infoblox data connector sample query updated
   - Infoblox workbook query changed

   Reason for Change(s):
   - Performance issue raised in ICM

   Version Updated:
   - NA
   
   Testing Completed:
   - With limited available data, Need to be tested in customers workspace

   Checked that the validations are passing and have addressed any issues that are present:
   - Validations passed

----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/Infoblox NIOS/Parser/Infoblox_allotherdhcpdTypes.txt [hunk 1]
```diff
@@ -20,14 +22,14 @@
 // This parser assumes the raw log are formatted as follows:
 // 
 //      May 13 12:05:52 10.0.0.0 dhcpd[30174]: DHCPDISCOVER from 0a:0b:0c:0d::0f via eth2 TransID 5daf9374: network 10.0.0.0/24: no free leases
-//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020.
-let datasource = (_GetWatchlist('InfobloxDevices') | project Computer);
-let RawData = Syslog
-    | where Computer in (datasource)// and Facility == "local6"
+//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020
+let datasource = (_GetWatchlist('Sources_by_SourceType')| project Source);
```
[the remark this answers is not in the document]
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_allotherdhcpdTypes.txt:26]: I have changed the query to use Searchkey

File: Solutions/Infoblox NIOS/Parser/Infoblox_allotherdhcpdTypes.txt [hunk 2]
```diff
@@ -20,14 +22,14 @@
...
+let RawData = Syslog 
+    | where Computer in (datasource)
     | extend Parser = extract_all(@"^(\d{2}\-[a-zA-Z]{3}\-\d{4}\s[0-9\.\:]+)?\s?([a-zA-Z-_]+)(\s|\:)?(.*)", dynamic([1,2,3,4]), SyslogMessage)[0]
     | extend responseTime = todatetime(Parser[0]),
             Log_Type = tostring(Parser[1]),
             RawData_subString = tostring(Parser[3])
```
[the remark this answers is not in the document]
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_allotherdhcpdTypes.txt]: Remove RawData_subString from the query

File: Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt [hunk 3]
```diff
@@ -20,14 +21,14 @@
 // This parser assumes the raw log are formatted as follows:
 // 
 //      May 13 12:05:52 10.0.0.0 dhcpd[30174]: DHCPDISCOVER from 0a:0b:0c:0d::0f via eth2 TransID 5daf9374: network 10.0.0.0/24: no free leases
-//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020.
-let datasource = (_GetWatchlist('InfobloxDevices') | project Computer);
-let RawData = Syslog
-    | where Computer in (datasource)// and Facility == "local6"
+//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020
```
[the remark this answers is not in the document]
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt:24]: Changed the query to use only syslogmessage filter and moved to line 28

File: Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json [hunk 4]
```diff
@@ -8,30 +8,30 @@
      { 
          "metricName": "Total data received", 
          "legend": "InfobloxNIOS", 
-         "baseQuery": "Infoblox_all" 
+         "baseQuery": "Infoblox_consolidated"
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json:11]: Question to @ms-sapangoel : Do we expect analysts to use **Infoblox_consolidated**? I would assume **Infoblox** would be a better choice for the primary parser.

File: Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json [hunk 5]
```diff
@@ -8,30 +8,30 @@
...
-               "lastDataReceivedQuery": "Infoblox_all \n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)" 
+               "lastDataReceivedQuery": "Infoblox_consolidated \n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)" 
             } 
         ], 
              "connectivityCriterias": [ 
                 { 
                     "type": "IsConnectedQuery", 
                     "value": [ 
-                    "Infoblox_all \n      | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(7d)" 
+                    "Infoblox_consolidated \n | where TimeGenerated > ago(3d)\n |take 1 \n | project isConnected = true"
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json:34]: @ms-sapangoel : are we standardizing on 3 days as the connection testing period? It is important to be consistent here.

File: Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt [hunk 6]
```diff
@@ -1,13 +1,14 @@
...
+// Last Updated:    20/05/2022
 // Comment:         Updated to only parse the MSG (RFC3164) part of the Syslog message. Excluded the Header.
 //
 // DESCRIPTION:
-// This parser takes raw Infoblox NIOS DHCP and DNS logs from a Syslog data stream and parses the data into a normalized schema
+// This parser comprised of function - Infoblox_allotherdnstypes
 //
 // USAGE:
+// USAGE:
 // 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window.
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt:12]: The usage instructions are not needed here and confusing/incorrect (as deploying any one function would not help). True for all parsers.

File: Solutions/Infoblox NIOS/Parser/Infoblox_consolidated.txt [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// Infoblox NIOS logging formats: https://docs.infoblox.com/display/NAG8/Using+a+Syslog+Server
+//
+// LOG SAMPLES:
+// This parser assumes the raw log are formatted as follows:
+// 
+//      May 13 12:05:52 10.0.0.0 dhcpd[30174]: DHCPDISCOVER from 0a:0b:0c:0d::0f via eth2 TransID 5daf9374: network //10.0.0.0/24: no free leases
+//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA //serial 9191919. Zone version is now 0202020.
+union isfuzzy=true
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Parser/Infoblox_consolidated.txt:25]: Use isfuzzy=false. You are hiding errors as such. isfuzzy=true should be used only for a query on multiple tables you are not sure one of exists. True for all union parsers in this batch.

File: Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+// 
+//      May 13 12:05:52 10.0.0.0 dhcpd[30174]: DHCPDISCOVER from 0a:0b:0c:0d::0f via eth2 TransID 5daf9374: network 10.0.0.0/24: no free leases
+//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020
+let datasource = (_GetWatchlist('Sources_by_SourceType')| where SearchKey == 'InfobloxNIOS' | project Source);
+Syslog
+    | where Computer in (datasource)
+    | where ProcessName == "dhcpd" and SyslogMessage has "DHCPACK"
+    | extend dhcpack = extract_all(@"(\s\w+\s(\S+))\sto\s(\S+)\svia\s(\S+)(\srelay\s(\S+))?(\slease-duration\s(\d+))?(\s\(([a-zA-Z]+)\))?(\suid\s(\S+))?", dynamic([1,2,3,4,5,6,7,8,9,10,11]), SyslogMessage)[0]
+    | extend IPAddress = tostring(dhcpack[1]), 
+        Client_MAC_Address = tostring(dhcpack[2]),
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt:31]: @ms-sapangoel : since we are creating a new version of the parsers anyways which is a breaking change, wouldn't it make sense to use normalized field names?

File: Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+// LOG SAMPLES:
+// This parser assumes the raw log are formatted as follows:
+// 
+//      May 13 12:05:52 10.0.0.0 dhcpd[30174]: DHCPDISCOVER from 0a:0b:0c:0d::0f via eth2 TransID 5daf9374: network 10.0.0.0/24: no free leases
+//      May 13 12:05:52 10.1.1.1 named[11325]: zone voip.abc.com/IN: ZRQ applied transaction 0101010 with SOA serial 9191919. Zone version is now 0202020
+let datasource = (_GetWatchlist('Sources_by_SourceType')| where SearchKey == 'InfobloxNIOS' | project Source);
+Syslog
+    | where Computer in (datasource)
+    | where ProcessName == "dhcpd" and SyslogMessage has "DHCPACK"
+    | extend dhcpack = extract_all(@"(\s\w+\s(\S+))\sto\s(\S+)\svia\s(\S+)(\srelay\s(\S+))?(\slease-duration\s(\d+))?(\s\(([a-zA-Z]+)\))?(\suid\s(\S+))?", dynamic([1,2,3,4,5,6,7,8,9,10,11]), SyslogMessage)[0]
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt:29]: I am not sure, but I think that somem parser regex, such as this one, assume no Syslog header, while others, such as "allother" assume syslog header.

File: Solutions/Infoblox NIOS/Workbooks/Infoblox-Workbook-V2.json [hunk 10]
```diff
@@ -77,7 +77,7 @@
               "multiSelect": true,
               "quote": "'",
               "delimiter": ",",
-              "query": "_GetWatchlist('InfobloxDevices')| project Computer",
+              "query": "_GetWatchlist('Sources_by_SourceType')| project Source",
```
Comment (oshezaf) [inline Solutions/Infoblox NIOS/Workbooks/Infoblox-Workbook-V2.json:80]: You did not fix this one.

**Review by oshezaf — CHANGES_REQUESTED** [review]
More commenrs.

Also:

- Since you removed the InfobloxNIOS parser, I think you need to update the Infoblox analytic rules. Actually it makes sense to update them for performance reasons as well.
- @ms-sapangoel : there are a few questions for you in the comments.

**Review by oshezaf — APPROVED** [review]

File: Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt (hunk 6 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_allotherdnsTypes.txt]: usage instruction is removed

File: Solutions/Infoblox NIOS/Parser/Infoblox_consolidated.txt (hunk 7 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_consolidated.txt]: isfuzzy=true is removed. Default is isfuzzy=false

File: Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt (hunk 8 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt]: I have updated normalized field names to parser functions

File: Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt (hunk 9 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt:23]: I don't have data to test this function.

File: Solutions/Infoblox NIOS/Workbooks/Infoblox-Workbook-V2.json (hunk 10 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Workbooks/Infoblox-Workbook-V2.json]: fixed

File: Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt (hunk 9 above)
Comment (oshezaf) [reply Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt:23]: Let me rephrase. I am quite sure that the parser is inconsistent and will not work on some event types. Did you use the data [here](https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/Syslog/InfobloxNIOS.json)?

Comment (rushriva) [reply Solutions/Infoblox NIOS/Parser/Infoblox_dhcpack.txt]: I have made changes to syslog headers and it's consistent for all parsers.

**Comment by rushriva** [comment]
@oshezaf I have made changes to syslog headers and it's consistent for all parsers.

File: Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json (hunk 5 above)
Comment (rushriva) [reply Solutions/Infoblox NIOS/Data Connectors/Connector_Syslog_Infoblox.json]: I had a discussion with Sapan on this and we are consistent with 3 days as the connection testing period.

**Review by oshezaf — APPROVED** [review]

**Review by NikTripathi — APPROVED** [review]
